mlflow/sagemaker/__init__.py

```diff
@@ -1747,6 +1769,704 @@ def _does_model_exist(model_name, sage_client):
+
+        .. code-block:: python
+            :caption: Python example
+
+            from mlflow.sagemaker import SageMakerDeploymentClient
+
+            client = SageMakerDeploymentClient("sagemaker:/us-east-1/arn:aws:123:role/assumed_role")
+            client.list_deployments()
+
+        .. code-block:: bash
+            :caption: Command-line example
+
+            mlflow deployments list --target sagemaker:/us-east-1/arn:aws:1234:role/assumed_role
+        """
+        import boto3
+
+        assume_role_credentials = _assume_role_and_get_credentials(
+            assume_role_arn=self.assumed_role_arn
+        )
+
+        sage_client = boto3.client(
+            "sagemaker", region_name=self.region_name, **assume_role_credentials
+        )
+        return sage_client.list_endpoints()["Endpoints"]
+
+    def get_deployment(self, name):
+        """
+        Returns a dictionary describing the specified deployment.
+
+        If a region name needs to be specified, the plugin must be initialized
+        with the AWS region in the ``target_uri`` such as ``sagemaker:/us-east-1``.
+
+        To assume an IAM role, the plugin must be initialized
+        with the AWS region and the role ARN in the ``target_uri`` such as
+        ``sagemaker:/us-east-1/arn:aws:1234:role/assumed_role``.
+
+        A :py:class:`mlflow.exceptions.MlflowException` will also be thrown when an error occurs
+        while retrieving the deployment.
+
+        :param name: Name of deployment to retrieve
+        :return: A dictionary that describes the specified deployment
+
+        .. code-block:: python
+            :caption: Python example
+
+            from mlflow.sagemaker import SageMakerDeploymentClient
+
+            client = SageMakerDeploymentClient("sagemaker:/us-east-1/arn:aws:123:role/assumed_role")
+            client.get_deployment("my-deployment")
+
+        .. code-block:: bash
+            :caption: Command-line example
+
+            mlflow deployments get --target sagemaker:/us-east-1/arn:aws:1234:role/assumed_role \\
+                --name my-deployment
+        """
+        import boto3
+
+        assume_role_credentials = _assume_role_and_get_credentials(
+            assume_role_arn=self.assumed_role_arn
+        )
+
+        try:
+            sage_client = boto3.client(
+                "sagemaker", region_name=self.region_name, **assume_role_credentials
+            )
+            return sage_client.describe_endpoint(EndpointName=name)
+        except Exception as exc:
+            raise MlflowException(
+                message=(f"There was an error while retrieving the deployment: {exc}\n")
+            )
+
+    def predict(self, deployment_name, df):
+        raise NotImplementedError("This function is not implemented yet.")
+
+    def explain(self, deployment_name, df):  # pylint: disable=unused-argument
+        raise NotImplementedError("This function is not implemented yet.")
```

Comment:
Can we remove these methods for now until they're implemented? This way, they won't show up in the docs.